internal/template/generator.go

```diff
@@ -40,8 +42,18 @@ func NewGenerator(
 }
 
 func (g *generator) ProcessTemplate() *template.Template {
-	return template.Must(
-		template.New(filepath.Base(g.source)).Funcs(g.funcMap.FuncMap).ParseFiles(g.source))
+	const helpersPath = "source/_helpers.tpl"
+	if _, err := os.Stat(helpersPath); err == nil {
+		return template.Must(
+			template.New(filepath.Base(g.source)).Funcs(g.funcMap.FuncMap).ParseFiles(g.source, helpersPath))
+	} else if errors.Is(err, os.ErrNotExist) {
+		logrus.Warnf("template helpers file '%s' not found\n", helpersPath)
+		return template.Must(
+			template.New(filepath.Base(g.source)).Funcs(g.funcMap.FuncMap).ParseFiles(g.source))
+	} else {
+		logrus.Fatalf("got unexpected error when checking if helper file exists at path %s: %s\n", helpersPath, err)
+		panic(err)
+	}
```

Comment:
Please refrain from panic-ing in library code: [return descriptive errors instead](https://go.dev/doc/effective_go#errors).
In this case Fatalf already exits, so this panic call will never get called and can be removed.
More info about Go "Zen" can be found here: https://go-proverbs.github.io

Reply:
the `Fatalf` exiting was my understanding but the VSCode linter was still asking for a return. I've deleted the `panic()` line.

Thanks for the links 🙏